.github/workflows/end2end-test.yml

```diff
@@ -1,10 +1,15 @@
 name: End to end test
 
 on:
-  # Trigger on all pull requests
   pull_request:
-    branches:
-      - "*"
+    branches: [ main, develop ]
+    # Limit to certain PR event types since this action doesn't need to run for every commit
+    types: [ready_for_review, review_requested]
+
+  # When changes are pushed to special branches
+  push:
+    branches: [main, develop]
+
```

Comment:
@au2 Are you good with this change?

Reply:
ok. i think there is a path in this setup to go update develop after changes without end to end test run but hopefully people with request reviews after changes.

Reply:
Yea, I'm not too concerned since the end-to-end test will run against the `develop` branch after the PR is merged, and we'll get alerted if it fails.